feat: e2e test adaptation and Ethereum tests implementation

alenmestrov opened the pull request:
# Description

Updated e2e tests to include proxy contract functionalities.
Introduced Ethereum tests.

## Test plan
N/A

## Documentation update
N/A

miraclx commented on e2e-tests/config/config.json:
```diff
@@ -34,6 +34,15 @@
       }
+    },
+    {
+      "protocol": "evm",
```
`"ethereum"`, but along with https://github.com/calimero-network/core/pull/1142#discussion_r1995920234, can be renamed in a follow-up PR

miraclx commented on e2e-tests/src/protocol/evm.rs:
```diff
@@ -0,0 +1,69 @@
+    pub fn node_args(&self) -> Vec<String> {
+        println!("config: {:?}", self.config);
+        vec![
+            format!("context.config.evm.protocol=\"{}\"", "evm"),
```
why am I just realizing we have a duplication with this

```toml
[context.config.near]
signer = "relayer"
protocol = "near"
network = "testnet"
contract_id = "calimero-context-config.testnet"
```

the `protocol` field is redundant (already named in the table), we can fix in a separate PR

miraclx replied:
Filed in #1149.

alenmestrov commented:
> Looks good generally, but I think we are missing `scripts/ethereum/deploy-devnet.sh` script and there is a lot of `println!` in the code

Yes, I cleaned it now and fixed it based on the comments and discussions we had, lets see how it behaves now
